src/main/java/com/github/terma/jenkins/githubprcoveragestatus/CompareCoverageAction.java

```diff
@@ -74,10 +87,29 @@ public void perform(
             return;
         }
 
+
         buildLog.println(BUILD_LOG_PREFIX + "start");
 
-        final String gitUrl = Utils.getGitUrl(build, listener);
-        final int prId = Utils.gitPrId(build, listener);
+        String gitUrl;
+        int prId;
+        try {
```

Comment:
`Utils.get*` throws exceptions in all cases, so... I dislike try/catch for control logic just as much, but didn't want to change too much, do we want to refactor `Utils` to return `null`?

I agree on the removing the setting, passing `scmVars` implies you opt-in to branch discovery.